core/objects.go

```diff
@@ -157,21 +158,69 @@ type Registration struct {
 	CreatedAt time.Time `json:"createdAt"`
 }
 
+// A type used to implement sort.Interface on a slice of AcmeURL pointers
+type contactList []*AcmeURL
+
+func (c contactList) Len() int { return len(c) }
+func (c contactList) Swap(i, j int) {
+	c[i], c[j] = c[j], c[i]
+}
+func (c contactList) Less(i, j int) bool {
+	return (*c[i]).String() < (*c[j]).String()
+}
+
 // MergeUpdate copies a subset of information from the input Registration
-// into this one.
-func (r *Registration) MergeUpdate(input Registration) {
+// into this one. It returns true if an update was performed and the base object
+// was changed, and false if no change was made.
+func (r *Registration) MergeUpdate(input Registration) (changed bool) {
 	// Note: we allow input.Contact to overwrite r.Contact even if the former is
 	// empty in order to allow users to remove the contact associated with
 	// a registration. Since the field type is a pointer to slice of pointers we
 	// can perform a nil check to differentiate between an empty value and a nil
 	// (e.g. not provided) value
 	if input.Contact != nil {
-		r.Contact = input.Contact
+		// If there is no existing contact slice, or the contact slice lengths
+		// differ, then the input is new and the base contact is changed
+		if r.Contact == nil || len(*input.Contact) != len(*r.Contact) {
+			r.Contact = input.Contact
+			changed = true
+		}
+
+		// Sort the two contact slices - coerce to `contactList` to satisfy
```

Comment:
I think this makes an assumption about client JSON serialization behavior that won't hold, since we know this is a relatively expensive operation to do at the SQL layer why not try to prevent it before then?